base/set.jl

```diff
@@ -625,10 +625,14 @@ end
 function _replace!(new::Callable, res::AbstractArray, A::AbstractArray, count::Int)
     c = 0
     if count >= length(A) # simpler loop allows for SIMD
-        for i in eachindex(A)
-            @inbounds Ai = A[i]
-            y = new(Ai)
```

Comment:
Afaict this is to make sure new runs with bounds check and you should keep it.

Reply:
Thanks, I didn't think about that.